lib/Sema/TypeCheckDeclPrimary.cpp

```diff
@@ -2147,6 +2147,31 @@ class DeclChecker : public DeclVisitor<DeclChecker> {
           } else {
             VD->diagnose(diag::property_does_not_override, isClassContext)
                 .highlight(OA->getLocation());
+            //If the override is a property-decl, check if there exists
+            //a no-args func-decl in the base class, if yes, provide a fix-it.
+            auto declName = VD->getBaseName();
```

Comment:
A couple of this here:

- There is no need to use constraint system for this. You need to extract nominal type (i.e. struct/enum/class) where it's declared via `DC->getSelfNominalTypeDecl()` - using this declaration it possible to lookup members via `lookupDirect` method.
-  The information about number of parameters and result type could be extracted from `FuncDecl`